quinn-proto/src/packet.rs

```diff
@@ -138,6 +212,34 @@ impl PartialDecode {
             if allow_coalesced { Some(bytes) } else { None },
         ))
     }
+
+    fn get_packet_number(
+        buf: &mut io::Cursor<BytesMut>,
+        pn_key: &PacketNumberKey,
+        sample_offset: usize,
+    ) -> Result<PacketNumber, PacketDecodeError> {
+        if buf.get_ref().len() < sample_offset + pn_key.sample_size() {
```

Comment:
Did you miss this bounds check (which I added in my last revision)? This seems to cover all the cases you comment on.

Reply:
I did miss this, because I was looking for something that adjusts the sample offset and continues rather than bailing out, per the TLS draft. The current state of affairs appears to assume that valid long packets will always have payloads extending at least 20 bytes from the start of the packet number, and it's not clear to me that that holds.